cockatrice/src/deck_loader.cpp

```diff
@@ -108,3 +110,72 @@ DeckLoader::FileFormat DeckLoader::getFormatFromName(const QString &fileName)
     }
     return PlainTextFormat;
 }
+
+bool DeckLoader::saveToStream_Plain(QTextStream &out)
+{
+    // deck header
+    if(!getName().isEmpty())
+        out << "// " << getName() << "\n\n";
+    if(!getComments().isEmpty())
+    {
+        QStringList commentRows = getComments().split(QRegExp("\n|\r\n|\r"));
+        foreach(QString row, commentRows)
+            out << "// " << row << "\n";
+        out << "\n";
+    }
+
+    // loop zones
+    for (int i = 0; i < getRoot()->size(); i++) {
```

Comment:
I typically like to use `foreach`, too. In this case it's possible, but it makes the code even harder to understand because the object returned by getRoot() is both a `InnerDecklistNode*` and a `QList<AbstractDecklistNode *>`, and we need to iterate over the QList and then `dynamic_cast` each returned abstract object back into a `InnerDecklistNode*` in order to loop over its childs:

```
    // loop zones
    InnerDecklistNode *rootNode = getRoot();
    foreach(AbstractDecklistNode *abstractNode, *rootNode)
    {
        const InnerDecklistNode *node =
            dynamic_cast<InnerDecklistNode *>(abstractNode);
```